### PR TITLE
Improve new user website navigation

### PR DESCRIPTION
**Pull request by starkos**
**What does this PR do?**

Make a few improvements to the website navigation:

- Add GitHub & Twitter links to the navbar
- Clean up and improve footer links
- Improve the flow of the intro pages
- Fix broken link on Building Premake
- Use a proper favicon
- Add OS icons on download page

**How does this PR change Premake's behavior?**

Website only; no Premake code changes.

**Anything else we should know?**

See #1547 for all the website setup discussion.

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -3,11 +3,17 @@ module.exports = {
...
+	],
 	onBrokenLinks: 'throw',
 	onBrokenMarkdownLinks: 'throw',
-	favicon: 'img/premake-logo.png', // FIXME: make actual favicon.ico file
+	favicon: 'img/favicon.ico',
 	organizationName: 'premake',
 	projectName: 'premake.github.io',
+	customFields: {
+		latestVersion5: '5.0.0-alpha15'
+	},
```
Comment (KyrietS) [inline website/docusaurus.config.js:16]: I'm afraid this field is not used anywhere. I guess you wanted to extract it in `download.js` page to replace this hardcoded constant?

```js
const LATEST_VERSION = '5.0.0-alpha15';
```

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -3,11 +3,17 @@ module.exports = {
 	tagline: 'Powerfully simple build configuration',
 	url: 'https://premake.github.io/',
 	baseUrl: '/',
+	scripts: [
+		'https://use.fontawesome.com/dd1c9cd9ff.js'
+	],
```
Comment (KyrietS) [inline website/docusaurus.config.js:8]: We could use [React Font Awesome](https://fontawesome.com/how-to-use/on-the-web/using-with/react) and `import` necessary icons in pages that need them. I'd leave it as it is for now. It's a place for improvement.

**Review by KyrietS — CHANGES_REQUESTED** [review]
Great commit! 😃 
Please address my first comment about unused field. The other one is just a side note :-)

File: website/docusaurus.config.js (hunk 1 above)
Comment (starkos) [reply website/docusaurus.config.js]: Nice catch, thanks. I went back and forth on the best way to set this up; missed this in my review.

File: website/docusaurus.config.js (hunk 2 above)
Comment (starkos) [reply website/docusaurus.config.js:8]: I started with RFA actually, but found it easier to just work with the class names. And there really isn't other way to do it for the navbar, as far as I could tell. You can only list class names there.

**Review by KyrietS — APPROVED** [review]

File: website/docusaurus.config.js (hunk 2 above)
Comment (KyrietS) [reply website/docusaurus.config.js:8]: Oh, you're right about the navbar. It's ok then 👍

**Pull request merged**